### PR TITLE
Update contributing

### PR DESCRIPTION
**Pull request by grd**

### CODE REVIEW

**Comment by elbing** [comment]
LGTM

**Comment by elbing** [comment]
Ok Gerard, you can merge it. Thanks!

**Comment by grd** [comment]
Okay, so I have to merge it, I thought that was done automatically. Let me update it with what we did have yesterday.

**Comment by elbing** [comment]
No I was wrong. It must merged for someone with permissions.

**Comment by sevki** [comment]
Hey @grd! Great to have you back :)

I want to see something that looks more like the original [document](https://github.com/Harvey-OS/harvey/blob/master/CONTRIBUTING.md) I adopted 

Something like [this](https://gist.github.com/sevki/20d4b841be68c0270d14)
### Submitting a Patch
Harvey uses Github Pull Requests to accept contributions. 
1. Clone the repo `git clone https://github.com/Harvey-OS/harvey.git`
2. Check out a feature branch you will work on by `git checkout -b feature-name` for instance for #70 issue, @keedon selected the branch name `statscrash`
3. Make changes
4. Commit with a descriptive message, like so
```
$ git commit -m "A brief summary of the commit
>
> A paragraph describing what changed and its impact."
```
take a look at @keedon's [commit message](https://github.com/keedon/harvey/commit/09fe3a21fa8b42088bc8ad83287928e9e7cc96ef) for the same issue for a good example of a descriptive commit message.
5. Fork the repo (only once)
harvey-os_harvey__a_fresh_take_on_plan_9
6. Add the repo as a remote (every time you clone the repo)
`git remote add yourname https://github.com/yourusername/harvey.git`
7. Push your branch to your forked repo like so `git push yourname feature-name`
8. Create a pull request by going to `https://github.com/yourusername/harvey/pull/new/feature-name` or clicking the PR button 
sevki_harvey__a_64_bit_operating_system_based_on_plan_9_from_bell_labs_and_nix__under_gpl
9. Add details of what you have worked on and what your intentions are. 
comparing_harvey-os_master___sevki_travis-trials_ _harvey-os_harvey

### Iterating on your pull request Iterate on your work. 

- A Github PR or a Gerrit CL roughly serve the same purpose, they are feedback loops for people to give and for you to iterate on your work, so be ready to repeat steps 3, 4 and 7. 
- When iterating on your work continue committing to the same branch and push the changes up to your fork. Github will track the changes and update the PR accordingly. 
- Most of the time while you have submitted your PR, someone else will have merged something to master, at which point your branch will have been outdated, GitHub provides a convenient way of updating your branch right from your pull request.
build_files_by_sevki_ _pull_request__53_ _harvey-os_harvey
- When you click that button, GitHub will update your branch, at which point you will have to `git pull yourname feature-name` and update your local repo before committing more changes.
- If your changes are conflicting with something that was merged into the branch, you will have to resolve the changes manually and submit the changes. 

#### More information
- [How to use Pull Requests](http://help.github.com/pull-requests/)
- [GitHub Flow](https://guides.github.com/introduction/flow/)
- [Hub: A command line application for GH Flow](https://hub.github.com)
- [Video: Pull Requests](https://www.youtube.com/watch?v=81uKcXZoQ2A)
- [Video: Workflows](https://www.youtube.com/watch?v=EwWZbyjDs9c) 
- ...

Maybe you can use [this](https://gist.github.com/sevki/20d4b841be68c0270d14) as a jumping point for adding some changes. 
More information section should give more love to rebasing, merging and syncing. Just the submitting a patch section is a massive portion of the document, so try to find good documentation and link to it rather than trying to explain it yourself and link it to them :)

**Comment by grd** [comment]
LGTM, I will update the doc with it.

**Comment by sevki** [comment]
pictures seems to be already stored at github. and this is the [raw document](https://gist.githubusercontent.com/sevki/20d4b841be68c0270d14/raw/fd8cd504a5b29f3a81abcbff34b1e12292b5ecc0/Submitapatch.md)

**Comment by grd** [comment]
Okay, there is something I have to do now, in a couple of hours I submit the patch.

**Comment by elbing** [comment]
It would be very explainive. I'm glad you both could make this better. Perhaps we could make new programmers life easier to taste and improve Harvey.

Thanks!

**Comment by grd** [comment]
Markdown numbering went wild and requires a small update...

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,47 +15,58 @@ Here are some important resources:
...
+## Submitting a patch
 
-Please send a [Pull Request to Harvey OS](https://github.com/Harvey-OS/harvey/pull/new/master) with a clear list of what you've done (read more about [pull requests](http://help.github.com/pull-requests/) and [Github Flow](https://guides.github.com/introduction/flow/)).
-
-Some of our contributors like using [hub](http://hub.github.com) for sending pull requests from the command line, an example workflow for contributing to harvey would look like (after you've [installed hub](http://hub.github.com))
-``` 
-$ git clone git@github.com:Harvey-OS/harvey.git
-$ cd harvey
+Harvey uses Github Pull Requests to accept contributions.
+1. Clone the repo `git clone https://github.com/Harvey-OS/harvey.git`. It is also possible to use git instead of https when you have an SSH public key stored in Github. Then you clone the repo with `git clone git@github.com/Harvey-OS/harvey.git`. This makes later submissions easier. For the rest of this manual we assume to use https.
```
Comment (sevki) [inline CONTRIBUTING.md]: missing line break before

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -15,47 +15,67 @@ Here are some important resources:
...
+3.  Make changes
+
+4.  Commit with a descriptive message, like so
+    ```
     $ git commit -m "A brief summary of the commit
     >
     > A paragraph describing what changed and its impact."
+    ```
+    take a look at @keedon's [commit message](https://github.com/keedon/harvey/commit/09fe3a21fa8b42088bc8ad83287928e9e7cc96ef) for the same issue for a good example of a descriptive commit message.
+    You can also use graphical git tools such as `git cola` which make commiting changes easier.
```
Comment (sevki) [inline CONTRIBUTING.md]: `git cola` is not a distributed with git and you haven't provided a link for it. However `git gui` is and it works cross platform so would you mind swapping out `git cola` for `git gui`.

you could maybe mention in the more information section [Githubs Desktop Client](https://desktop.github.com/) for osx and windows which supports pull requests.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -15,47 +15,67 @@ Here are some important resources:
...
+- When you click that button, GitHub will update your branch, at which point you will have to `git pull yourname feature-name` and update your local repo before committing more changes.
+- If your changes are conflicting with something that was merged into the branch, you will have to resolve the changes manually and submit the changes.
+
+#### More information
+
+- [How to use Pull Requests](http://help.github.com/pull-requests/)
+- [GitHub Flow](https://guides.github.com/introduction/flow/)
+- [Hub: A command line application for GH Flow](https://hub.github.com)
+- [Video: Pull Requests](https://www.youtube.com/watch?v=81uKcXZoQ2A)
+- [Video: Workflows](https://www.youtube.com/watch?v=EwWZbyjDs9c)
```
Comment (sevki) [inline CONTRIBUTING.md]: Would you mind adding some information here about rebasing and keeping your branch in sync?

Comment (sevki) [inline CONTRIBUTING.md]: rebase tutorials
https://www.youtube.com/watch?v=W39CfI3-JFc
https://www.youtube.com/watch?v=SxzjZtJwOgo

**Comment by sevki** [comment]
LGTM

**Pull request merged**